tests/utils/generate_synthetic_labels.cpp

```diff
@@ -12,27 +12,27 @@ namespace po = boost::program_options;
 class ZipfDistribution
 {
   public:
-    ZipfDistribution(int num_points, int num_labels)
+    ZipfDistribution(uint64_t num_points, uint32_t num_labels)
         : num_labels(num_labels), num_points(num_points),
           uniform_zero_to_one(std::uniform_real_distribution<>(0.0, 1.0))
     {
     }
 
-    std::unordered_map<int, int> createDistributionMap()
+    std::unordered_map<uint32_t, uint32_t> createDistributionMap()
```

Comment:
int has become uint32_t here. make sure this preserves correctness?

Reply:
The expected behavior is that both the key and value are non-negative integer values: the key is the integer value for the generated label, and the value is the number of occurrences of that label according to the distribution.

However, line 51 (`distribution_map[it->first] -= 1;`) would have some unwanted behavior if the number of expected occurrences were somehow set to 0. Refactoring this.